bebraven/platform
# SECURITY: Require 1 uppercase, 1 lowercase, 1 digit, 1 symbol for passwords

### Description
Task: https://app.asana.com/0/1201131148207877/1201543544397148

Don't allow super insecure passwords going forward. This approach and the regex to check complexity is taken from [Devise documentation](https://github.com/heartcombo/devise/wiki/How-To:-Set-up-simple-password-complexity-requirements). Note that I enhanced the regex to allow more special characters.

Also, I setup a Honeycomb trigger to alert when there is possible brute force attack:
https://ui.honeycomb.io/braven/datasets/platform.braven.org/triggers/dsL8katLvfQ?view=/braven/datasets/platform.braven.org/triggers

Follow-up task for another attack vector that this PR does nothing to solve for: https://app.asana.com/0/1201131148207877/1201607387698620

### Motivation and Context
Without this change, it's likely that at some point someone's account will be compromised simply b/c they used a password like "password". We're not trying to enforce super secure passwords, we're not a bank, but they should have some bare level of complexity/strength so that a brute force attack would require enough guesses to trigger alerts.

We should do this now, after only 1 semester of using this new user database, in order to avoid a painful "everyone needs to reset their password" exercise down the road with many times more users in the database.

### Test plan
1. login with existing account just to make sure it doesn't impact existing credentials, only new ones.
1. Reset password flow. Try the following:
    1. too short
    1. mismatched
    1. not complex enough
    1. success
    1. login with new password
1. Register new account flow. Try the following:
    1. Make sure and run sync on Enrolled Participant first, make sure the User record is created with an empty password.
    1. too short
    1. mismatched
    1. not complex enough
    1. success
    1. logout
    1. log back in with password

### Screenshots (if appropriate):

Register Account (Desktop)
![image](https://user-images.githubusercontent.com/5596986/147833990-634e61c8-1b2d-4bcb-8230-84413f75b3d8.png)

Register Account (Mobile)
<img width="250" src="https://user-images.githubusercontent.com/5596986/147833984-5557b666-afd8-4a8e-acf9-8f44572bc901.png">

Reset password (Desktop)
![image](https://user-images.githubusercontent.com/5596986/147833273-6b5d5e73-6e96-4eb0-b0f6-eb0ff00eef00.png)

Reset password (Mobile)
<img width="250" src="https://user-images.githubusercontent.com/5596986/147833276-1906a892-d10c-4fa7-b34e-09d35550cdd8.png">

### Types of changes




### Checklist:
